source/slang/slang-ir-inst-defs.h

```diff
@@ -195,6 +195,9 @@ INST(RTTIType, rtti_type, 0, 0)
 INST(RTTIHandleType, rtti_handle_type, 0, 0)
 INST(TupleType, tuple_type, 0, 0)
 
+// A type that identifies it's contained type as being emittable as `spirv_literal.
+INST(SPIRVLiteralType, spirvLiteralType, 1, 0)
+
```

Comment:
From my own attempts to look into it, the semantics of `spirv_literal` in GLSL/glslang appear to be something like:

> "When encoding SPIR-V words for a call to the `spirv_instruction` function this parameter belongs to, encode the argument for this parameter as a literal value in the SPIR-V binary form, rather than as the `<id>` of another instruction."

Given that understanding, it seems that neither this type-based approach (`SPIRVLiteral<int>`) nor an encoding as a rate (`@SPIRVLiteral int`) are semantically correct.

I believe the semantically-correct encoding would be to make this be `SPIRVLiteralDecoration`, a subtype of `Decoration`, and then attach that decoration to the `IRParam` in question directly (so *not* part of the type).

This PR works fine, so there is no immediate reason to change anything, but I want to note that in case we need/want to revisit this code.

Reply:
I did look at adding as a decoration originally. That such a decoration going in an IRParam would work fine, assuming there was an IRParam instruction. In this case we are adding a function forward declaration, so it has no body/blocks, so has no IRParam? 

Is the idea here to add a block/IRParams, and use some other mechanism to determine it has no body? Or am I missing something?